sdk/core/System.ClientModel/src/Internal/SSE/ServerSentEventReader.cs

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.ClientModel.Internal;
+
+// TODO: Different sync and async readers to dispose differently?
+internal sealed class ServerSentEventReader : IDisposable, IAsyncDisposable
+{
+    private Stream? _stream;
+    private StreamReader? _reader;
+
+    public int? LastEventId { get; private set; }
```

Comment:
This is part of the story around resuming the connection from the last received event in the case that the connection is lost before the full set of events has been delivered over the response content stream.  If we don't expose this to end users, we would need to implement reconnection as part of the type.

Reply:
Per offline discussion, we will not address requirements around resuming lost connections in this iteration of the work.